clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -305,13 +312,28 @@ public void completed(
           String message,
           Optional<LeaderFollowerStateType> leaderState) {
         // Use thread pool to handle the completion reporting to make sure it is not blocking the report.
-        if (isTopicPartitionIngesting(kafkaTopic, partition)) {
+        if (isTopicPartitionHosted(kafkaTopic, partition)) {
           getCompletionHandlingExecutor().submit(() -> {
             VeniceStoreVersionConfig config = getConfigLoader().getStoreConfig(kafkaTopic);
             config.setRestoreDataPartitions(false);
             config.setRestoreMetadataPartition(false);
             // Start partition consumption locally.
-            startConsumption(config, partition, leaderState);
+            try {
+              startConsumptionLocally(config, partition, leaderState);
+            } catch (Exception e) {
+              LOGGER.warn(
+                  "Caught exception when resuming ingestion of topic: {}, partition: {} in main process",
+                  kafkaTopic,
+                  partition,
+                  e);
+            } finally {
+              getMainIngestionMonitorService().setVersionPartitionToLocalIngestion(kafkaTopic, partition);
```

Comment:
(This is dependent on the answer to the prev comment) Shouldn't we be setting this config before invoking `startConsumptionLocally`?

Reply:
The idea here is that:
Before we start consumption on the main process successfully (or exceptionally if store is deleted in between, this only happens in weird EI env), the resource should still be considered as "handover/in transistion" state. If we set the state config before this is completed, pending Helix transition on the same resource will get executed on local and may lead to unexpected result.

Reply:
Thanks. That makes sense. I had a misunderstanding that child process and main process could end up ingesting at the same time. Confirmed offline that this is not possible.